accounts-db/src/accounts_db.rs

```diff
@@ -3944,21 +3988,50 @@ impl AccountsDb {
                     } else {
                         // Hold onto the index entry arc so that it cannot be flushed.
                         // Since we are shrinking these entries, we need to disambiguate storage ids during this period and those only exist in the in-memory accounts index.
-                        index_entries_being_shrunk.push(Arc::clone(entry.unwrap()));
```

Comment:
I don't think we can remove this line. 
@jeffwashington Do you know the reason that we removed this line?

Reply:
this can be removed and should. We can remove `index_entries_being_shrunk` from the outer struct as well. This was an optimization that is a good idea, but made unnecessary in this pr. The entries we care about are already held in the in-mem index. The ones not in the in-mem index yet we do not want to hold onto. We'll load them when we write the new storage.

Reply:
ok. I will remove it from the outer struct.

Reply:
https://github.com/anza-xyz/agave/pull/2689/commits/6677ced9dc428774353d8de1d01e8585585d9836